src/test/ReactTestUtils.js

```diff
@@ -309,6 +309,44 @@ var ReactTestUtils = {
     return new ReactShallowRenderer();
   },
 
+  jasmineMatchers: {
+    /**
+    * Expect function to perform a console.warn.
+    * If `expectation` is a number, we expect the function to make `expectation` calls to warn.
+    * If `expectation` is a string, we expect that string to appear in the warn output.
+    **/
+    toWarn: function(expectation) {
```

Comment:
Let's make this work more like toThrow where it'll set a message - https://github.com/jasmine/jasmine/blob/1_3_x/src/core/Matchers.js#L320-L346